src/CommandHandler/Wishlist/RemoveSelectedProductsFromWishlistHandler.php

```diff
@@ -0,0 +1,84 @@
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\CommandHandler\Wishlist;
+
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\AddWishlistProduct;
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\RemoveSelectedProductsFromWishlist;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class RemoveSelectedProductsFromWishlistHandler implements MessageHandlerInterface
+{
+    private ProductVariantRepositoryInterface $productVariantRepository;
+
+    private EntityManagerInterface $wishlistProductManager;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private int $itemsProcessed = 0;
+
+    public function __construct(
+        ProductVariantRepositoryInterface $productVariantRepository,
+        EntityManagerInterface $wishlistProductManager,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator
+    ) {
+        $this->productVariantRepository = $productVariantRepository;
+        $this->wishlistProductManager = $wishlistProductManager;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+    }
+
+    public function __invoke(RemoveSelectedProductsFromWishlist $removeSelectedProductsFromWishlistCommand): void
+    {
+        $this->removeSelectedProductsFromWishlist($removeSelectedProductsFromWishlistCommand->getWishlistProducts());
+
+        $this->addFlashMessage();
+    }
+
+    private function removeSelectedProductsFromWishlist(Collection $wishlistProducts): void
+    {
+        /** @var AddWishlistProduct $wishlistProduct */
+        foreach ($wishlistProducts as $wishlistProduct) {
+            if ($wishlistProduct->isSelected()) {
+                $this->removeProductFromWishlist($wishlistProduct);
+            }
+        }
+    }
+
+    private function removeProductFromWishlist(AddWishlistProduct $wishlistProduct): void
+    {
+        $productVariant = $this->productVariantRepository->find($wishlistProduct->getWishlistProduct()->getVariant());
+
+        if (null === $productVariant) {
+            throw new NotFoundHttpException();
+        }
+
+        $this->wishlistProductManager->remove($wishlistProduct->getWishlistProduct());
+        ++$this->itemsProcessed;
+    }
+
+    private function addFlashMessage(): void
+    {
+        if (0 < $this->itemsProcessed) {
```

Comment:
`else` is not required here: https://dev.to/tadeubdev/object-calisthenics-with-php-en-2h4p — this will simplify the syntax.